app/plugin.go

```diff
@@ -182,6 +187,72 @@ func (a *App) InitPlugins(pluginDir, webappPluginDir string) {
 	a.SyncPluginsActiveState()
 }
 
+// SyncPlugins synchronizes the plugins installed locally
+// with the plugin bundles available in the file store.
+func (a *App) SyncPlugins() *model.AppError {
+	mlog.Info("Synching plugins with the file store")
+
+	a.Srv.PluginsLock.RLock()
+	pluginsEnvironment := a.Srv.PluginsEnvironment
+	a.Srv.PluginsLock.RUnlock()
+
+	availablePlugins, err := pluginsEnvironment.Available()
+	if err != nil {
+		return model.NewAppError("SyncPlugins", "app.plugin.sync.read_local_folder.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	for _, plugin := range availablePlugins {
+		pluginId := plugin.Manifest.Id
+
+		// Only handle managed plugins with .filestore flag file.
+		_, err := os.Stat(filepath.Join(*a.Config().PluginSettings.Directory, pluginId, managedPluginFileName))
+		if os.IsNotExist(err) {
+			mlog.Warn("Found unmanaged plugin. Ignoring in plugin sync with the filestore.", mlog.String("plugin", pluginId))
+		} else if err != nil {
+			mlog.Error("Error reading local plugin directoy. Skipped for sync.", mlog.String("folder", pluginId), mlog.Err(err))
+		} else {
+			mlog.Debug("Plugin Sync Uninstalling plugin locally", mlog.String("plugin", pluginId))
+			if err := a.removePluginLocally(pluginId); err != nil {
+				mlog.Error("Error uninstalling managed plugin during sync.", mlog.String("plugin", pluginId), mlog.Err(err))
+			}
+		}
+	}
+
+	// Install plugins from the file store.
+	exists, appErr := a.FileExists(fileStorePluginFolder)
+	if appErr != nil {
+		return model.NewAppError("SyncPlugins", "app.plugin.sync.check_filestore.app_error", nil, appErr.Error(), http.StatusInternalServerError)
+	}
+
+	if !exists {
+		return nil
+	}
+
+	fileStorePaths, appErr := a.ListDirectory(fileStorePluginFolder)
+	if appErr != nil {
+		return model.NewAppError("SyncPlugins", "app.plugin.sync.list_filestore.app_error", nil, appErr.Error(), http.StatusInternalServerError)
+	}
+
+	for _, path := range fileStorePaths {
+		if strings.HasSuffix(path, ".tar.gz") {
+			var reader filesstore.ReadCloseSeeker
+			reader, appErr = a.FileReader(filepath.Join("./", path))
```

Comment:
Out of curiosity, is the join of `./` here necessary?

Reply:
Not anymore, it was having a hard time with that path when use `a.ReadFile` before. It's good now. Removed✅